src/main/java/org/jabref/gui/entryeditor/fileannotationtab/FileAnnotationViewModel.java

```diff
@@ -31,7 +31,11 @@ private void setupContentProperties(FileAnnotation annotation) {
             this.content.set(annotation.getLinkedFileAnnotation().getContent());
             String annotationContent = annotation.getContent();
             String illegibleTextMessage = Localization.lang("The marked area does not contain any legible text!");
-            this.marking.set(annotationContent.isEmpty() ? illegibleTextMessage : annotationContent);
+            String markingContent = (annotationContent.isEmpty() ? illegibleTextMessage : annotationContent);
+            // remove newlines && hyphens before linebreaks
+            markingContent = new RemoveHyphenatedNewlinesFormatter().format(markingContent);
+            markingContent = new RemoveNewlinesFormatter().format(markingContent);
+            this.marking.set(markingContent);
```

Comment:
It would be nice if you could add some tests verifying the new behavior: create a `FileAnnotion` object by hand (not by parsing a pdf) which contains such problematic hyphens/newlines and check that they are removed, indeed.

Reply:
Thank you for your advice, I have added a unit test!